test/cql-pytest/test_allow_filtering.py

```diff
@@ -308,6 +307,45 @@ def table4(cql, test_keyspace):
 
 # Selecting from indexed table should using only clustering key should require filtering
 # Variation of #7608 found in #8991
-@pytest.mark.xfail(reason="Select from indexed table using only clustering key should require filtering. Issue #7608")
 def test_select_indexed_cluster(cql, table4):
     check_af_mandatory(cql, table4, 'c1 = 1 AND c2 = 1', lambda row: row.c1 == 1 and row.c2 == 1)
+
+# table5 contains an indexed table with 3 clustering columns.
+# used to test need_filtering() on indexed tables.
+@pytest.fixture(scope="module")
+def table5(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int, c1 frozen<list<int>>, c2 frozen<list<int>>, c3 int, PRIMARY KEY (p,c1,c2,c3))")
+    cql.execute(f"CREATE INDEX ON {table} (c3)")
+    cql.execute(f"INSERT INTO {table} (p, c1, c2, c3) VALUES (0, [1], [2], 0)")
+    cql.execute(f"INSERT INTO {table} (p, c1, c2, c3) VALUES (0, [1], [2], 1)")
+
+    everything = list(cql.execute(f"SELECT * FROM {table}"))
+    wait_for_index(cql, table, 'c3', everything)
+    yield (table, everything)
+    cql.execute(f"DROP TABLE {table}")
+
+# Test that implementation of need_filtering() from #8994 for indexes works correctly.
+# This implementation is a bit conservative - it might sometimes state
+# that filtering is needed when it isn't actually required, but at least it's safe.
+@pytest.mark.xfail(reason="Too conservative need_filtering() implementation")
+def test_select_indexed_cluster_three_keys(cql, table5, scylla_only):
```

Comment:
Thanks to the table5 fixture (which is a super useful pytest feature), you can easily break this long tests into multiple tests, each with a different set of xfail / cassandra_bug / comments. Right now you have a big test which is marked both xfail (fails on Scylla) and scylla_only (should be cassandra_bug, meaning fails on Cassandra) - which is strange. Better have several smaller tests, each of them failing on just one of the two implementation. Unless you really found a common bug in both implementations (which is also possible).